scrapper.py

```diff
@@ -21,43 +38,180 @@ class IncorrectNumberOfArticlesError(Exception):
     """
 
 
+def _clean_text(text):
+    return re.sub(r"[\n\t ]+", " ", text).strip()
+
+
+def _get_page(link):
+    headers = Headers().generate()
+    response = requests.get(link, headers=headers)
+    if not response.ok:
+        return None
+    return BeautifulSoup(response.text, "html.parser")
+
+
 class Crawler:
     """
     Crawler implementation
     """
+
     def __init__(self, seed_urls, max_articles: int):
-        pass
+        self.seed_urls = seed_urls
+        self.max_articles = max_articles
+        self.urls = []
 
     def _extract_url(self, article_bs):
-        pass
+        if not article_bs:
+            return
+        for node in article_bs.find_all("a", {"class": "file"}):
+            # ignore links to issues, only collect articles.
+            # this does not leave out any content, because issues
+            # are comprised of the same articles.
+            if "issue" in node["href"]:
+                continue
+            if len(self.urls) == self.max_articles:
```

Comment:
i have checked my code, this check duplication exists to avoid making unnecessary extra requests to the site. Without it, it is more likely the crawler will get throttled. I have tested this.